### ISSUE TITLE
ArgumentNullException in AstPythonConstant ctor

### ISSUE THREAD

**Issue by huguesv** [Microsoft/PTVS#3040]
Failing test is `TestDiscovererTests.TestCaseSubclasses`

`type` is `null`:

```
        public AstPythonConstant(IPythonType type, params LocationInfo[] locations) {
            Type = type ?? throw new ArgumentNullException(nameof(type));
            Locations = locations.ToArray();
        }
```

calling code is:
```
                            var mem = mod.GetMember(ctxt, node.Names[i].Name) ?? new AstPythonConstant(
                                _interpreter.GetBuiltinType(BuiltinTypeId.Unknown),
                                GetLoc(n)
                            );
```

```
	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Interpreter.Ast.AstPythonConstant.AstPythonConstant(Microsoft.PythonTools.Interpreter.IPythonType type, Microsoft.PythonTools.Analysis.LocationInfo[] locations) Line 25	C#	Symbols loaded.
>	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Interpreter.Ast.AstAnalysisWalker.Walk(Microsoft.PythonTools.Parsing.Ast.FromImportStatement node) Line 169	C#	Symbols loaded.
 	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Parsing.Ast.FromImportStatement.Walk(Microsoft.PythonTools.Parsing.Ast.PythonWalker walker) Line 79	C#	Symbols loaded.
 	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Parsing.Ast.SuiteStatement.Walk(Microsoft.PythonTools.Parsing.Ast.PythonWalker walker) Line 37	C#	Symbols loaded.
 	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Parsing.Ast.PythonAst.Walk(Microsoft.PythonTools.Parsing.Ast.PythonWalker walker) Line 69	C#	Symbols loaded.
 	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Interpreter.Ast.AstPythonModule.AstPythonModule(string moduleName, Microsoft.PythonTools.Interpreter.IPythonInterpreter interpreter, Microsoft.PythonTools.Parsing.Ast.PythonAst ast, string filePath) Line 113	C#	Symbols loaded.
 	Microsoft.PythonTools.Analysis.dll!Microsoft.PythonTools.Interpreter.Ast.AstPythonModule.FromStream(Microsoft.PythonTools.Interpreter.IPythonInterpreter interpreter, System.IO.Stream sourceFile, string fileName, Microsoft.PythonTools.Parsing.PythonLanguageVersion langVersion, string moduleFullName) Line 78	C#	Symbols loaded.
 	TestAdapterTests.dll!TestAdapterTests.TestDiscovererTests.GetTestCasesFromAst(string code, Microsoft.PythonTools.Analysis.PythonAnalyzer analyzer) Line 98	C#	Symbols loaded.
 	TestAdapterTests.dll!TestAdapterTests.TestDiscovererTests.TestCaseSubclasses() Line 185	C#	Symbols loaded.
 	[Native to Managed Transition]		Annotated Frame
 	[Managed to Native Transition]		Annotated Frame
```

**Comment by zooba** [comment]
Let's do initial reports without triaging, and then use those tags when we go through new issues formally.

**Comment by zooba** [comment]
Fixed as part of #3059
